ui/nuxeo-routing-behavior.js

```diff
@@ -98,19 +102,66 @@ export const RoutingBehavior = {
   _computeNavigateTo() {
     return function(...args) {
       if (this.router) {
-        const route = args[0];
+        const [route, ...params] = args;
         const baseUrl = this.router.baseUrl || '';
-        if (route.startsWith('/')) {
-          this.router.navigate(baseUrl + route);
+        let path;
+        if (typeof route === 'object') {
+          path = this._routeEntity(...args);
+        } else {
+          if (route.startsWith('/')) {
+            this.router.navigate(baseUrl + route);
+          }
+          if (!this.router[route]) {
+            console.error(`Could not navigate to a url for route ${route}`);
+          }
+          path = this.router[route].apply(this, params);
         }
-        if (!this.router[route]) {
-          console.error(`Could not navigate to a url for route ${route}`);
-        }
-        const params = Array.prototype.slice.call(args, 1);
-        this.router.navigate(this.router[route].apply(this, params));
+        this.router.navigate(path);
       } else {
         console.error('No router defined');
       }
     };
   },
+
+  _routeEntity(...args) {
+    if (args.length === 0) {
+      return;
+    }
+    const [obj, ...params] = args;
+    if (typeof obj !== 'object') {
+      throw new Error(`cannot resolve route: "${obj}" is not a valid entity object`);
+    }
+    let entityType = obj['entity-type'];
+    if (!entityType) {
+      // XXX Sometimes we don't have the entity type. For example, on nuxeo-document-storage, we were not storing it.
```

Comment:
can we fix this at `nuxeo-document-storage` level so we store the entity-type and add it in case it's missing on stored documents so we can do without this hack here ?

Reply:
We could certainly fix this for newly added documents, but I'm not quite sure how to fix this for already existing documents in a reliable way, since we just forward the `value` of the inner `iron-localstorage`. We do have a `get` method which we use on some elements, but I don't think it would cover all use cases, specially those that rely on data binding. 

Alternatively, we could try to do something with a new private property to hold the "fixe" set of documents, or we could try to run something when the value changes to fix the documents, but I'm not sure it is worth the effort?

Reply:
We can try to fix this on nuxeo-home, which is the only place by default where we generate URLs from docs stored on local storage, AFAIK. But I'm not sure I'd call this robust.